cyclonedx/output/json.py

```diff
@@ -151,7 +151,7 @@ def _recurse_specialise_component(self, bom_json: Dict[Any, Any], base_key: str
                 if not self.component_supports_author() and 'author' in bom_json[base_key][i].keys():
                     del bom_json[base_key][i]['author']
 
-                if self.component_version_optional() and bom_json[base_key][i]['version'] == "":
+                if self.component_version_optional() and bom_json[base_key][i].get('version', '') == "":
```

Comment:
this will raise an error,
if `bom_json[base_key][i]['version']` is unset.

example:
```py
x = dict()
if x.get('version', '') == "":
  del x['version']
```

Reply:
here is a fix: https://github.com/CycloneDX/cyclonedx-python-lib/pull/194#discussion_r829293005